src/bolos/cx_ec_domain.c

```diff
@@ -1399,6 +1484,80 @@ int cx_nid_from_curve(cx_curve_t curve)
   return nid;
 }
 
+EC_GROUP *cx_group_from_nid_and_curve(int nid, cx_curve_t cid)
+{
+  EC_GROUP *group = NULL;
+  /* create a generic curve if generic nid */
+  if (nid != 0) {
+    group = EC_GROUP_new_by_curve_name(nid);
+  } else {
+    group = cx_create_generic_curve(cid);
+  }
+  return group;
+}
+
+EC_GROUP *cx_create_generic_curve(cx_curve_t cid)
+{
+  /* Get curve domain to init the EC_GROUP */
+  const cx_curve_domain_t *domain;
+  if (NULL == (domain = cx_ecdomain(cid))) {
+    errx(1, "error when getting the curve domain");
+    return NULL;
+  }
+
+  BN_CTX *ctx;
+  EC_GROUP *curve;
+  BIGNUM *a, *b, *p, *order, *x, *y;
+  EC_POINT *generator;
+
+  /* Set up the BN_CTX  */
+  if (NULL == (ctx = BN_CTX_new())) {
+    errx(1, "error when setting up the BN CTX");
+    return NULL;
+  }
+
+  /* Set up the values for the various parameters */
+  if ((NULL == (a = BN_bin2bn(domain->a, domain->length, NULL))) ||
+      (NULL == (b = BN_bin2bn(domain->b, domain->length, NULL))) ||
+      (NULL == (p = BN_bin2bn(domain->p, domain->length, NULL))) ||
+      (NULL == (order = BN_bin2bn(domain->n, domain->length, NULL))) ||
+      (NULL == (x = BN_bin2bn(domain->Gx, domain->length, NULL))) ||
+      (NULL == (y = BN_bin2bn(domain->Gy, domain->length, NULL)))) {
+    errx(1, "error when setting up the BN CTX parameters");
+    return NULL;
```

Comment:
Returning `NULL` here prevent allocated variables (with `BN_bin2bn`) from being freed. This pattern is also present in error handling below.

I'd recommend adding an `error:` label before variables freeing at the end of the function and using `goto error;` instead of `return NULL;`